### DISCUSSION TITLE
Devnet Pre-Release 1.2.1alpha1

### DISCUSSION THREAD

**Discussion by lk86** [MinaProtocol/mina#9632] [Announcements]
The O(1) Labs team is proud to announce that we're moving onto the 1.2.1 range of releases in the anticipation of [1.2.0beta8](https://github.com/MinaProtocol/mina/releases/tag/1.2.0beta8) soon graduating to stable. This first 1.2.1 Alpha release [1.2.1alpha1](https://github.com/MinaProtocol/mina/releases/tag/1.2.1alpha1) includes the more comprehensive changes to improve graphQL/networking responsiveness as promised, as well as a new async VRF computation process to keep VRF evaluations off of the main thread (especially for block producers with many delegates). 

This alpha release includes all of the changes from [1.2.0beta8](https://github.com/MinaProtocol/mina/releases/tag/1.2.0beta8), including the new uptime tracking system, but it is not recommended to use this feature on devnet unless its for testing against your own backend. Otherwise you can expect the same workflow as using 1.2.0beta8 with no new breaking changes.

If you run into any issues, please include `Release: 1.2.1alpha1` in the name of any issues you create. Please report any issues you encounter [here](https://github.com/MinaProtocol/mina/issues/new/choose) or join the Discussion below!

### Stability Improvements and Bug Fixes:
- Async scheduler changes to improve responsiveness #9416 and related metrics #9545
- Stop restarting the Verifier every 15 minutes #9565 #9568

### New Features:
- Asynchronous VRF evaluation process #9409
- Change the docker entrypoint script so that the last few lines of log output are less likely to get cut off #9568
- Properly tag releases in CI and push them directly to minaprotocol/ #9566

### Archive Node:
- Update mina-extract-blocks to properly set the block_winner instead of always using creator_id #9567 (Thanks @kobigurk for making the change and @jrwashburn for bringing attention to this bug)

### Complete Changelog since 1.2.0beta8:
https://github.com/MinaProtocol/mina/compare/1.2.0beta8...1.2.1alpha1
For more information about the changes included in 1.2.0beta8, see the [previous release notes](https://github.com/MinaProtocol/mina/releases/tag/1.2.0beta8)

## Upgrading & Connecting

**Debian Repository:**
First, Set up and update the `alpha` Debian Repository
```
echo "deb [trusted=yes] http://packages.o1test.net stretch alpha" | sudo tee /etc/apt/sources.list.d/mina-alpha.list
echo -e "Package: mina-mainnet\nPin: release c=alpha\nPin-priority: 1" | sudo tee /etc/apt/preferences.d/99-mina-alpha
sudo apt-get update
```
Then, install the package(s) that you need:
**Daemon Package:** `sudo apt-get install -y mina-devnet=1.2.1alpha1-1f98b8b`
**Archive Node Package:** `sudo apt-get install -y mina-archive=1.2.1alpha1-1f98b8b`

### Docker Images:
**Daemon Image:** `minaprotocol/mina-daemon:1.2.1alpha1-1f98b8b-devnet`
**Archive Image:** `minaprotocol/mina-archive:1.2.1alpha1-1f98b8b`

### Sandbox Node: For testing in an isolated, single-node network without snarks
As of 1.2.0, the demo functionality is built into the standard daemon container, via the environment variable `RUN_DEMO=true`.
Just `docker run -it --name mina-demo -e RUN_DEMO=true minaprotocol/mina-daemon:1.2.1alpha1-1f98b8b-devnet` and go!

### Step by Step Guide:
Check out [our documentation](https://docs.minaprotocol.com/en/using-mina/connecting) for complete instructions on using this version to connect to Mainnet. *Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/seed-lists/devnet_seeds.txt`.

If you are running the correct version on the correct network, `mina client status` will show:
```
Chain id:     b6ee40d336f4cc3f33c1cc04dee7618eb8e556664c2b2d82ad4676b512a82418
Git SHA-1:    1f98b8be743e78993670e7b148be72c894b212e3
```

<hr /><em>This discussion was created from the release <a href='https://github.com/MinaProtocol/mina/releases/tag/1.2.1alpha1'>Devnet Pre-Release 1.2.1alpha1</a>.</em>

**Comment by GistKen** [comment]
I've been running the `1.2.1alpha1-1f98b8b-devnet` docker container for 36 hours without any issues so far.
Additionally, I've been running it on the mainnet with 36 hours without any issue so far.

**Comment by GistKen** [comment]
I think there is a potential memory leak in the releases (in general). I only started paying attention in this latest alpha release but I see the following memory usage for the mainnet containers that I migrated over to the alpha release:
Container 1 with uptime of 1 Hour: 7.5 GB
Container 2 with uptime of 44 Hours: 11.23 GB
Container 3 with uptime of 63 Hours: 12.23 GB

**Comment by GistKen** [comment]
Docker container: `minaprotocol/mina-daemon:1.2.1alpha1-1f98b8b-mainnet `
Node went offline prior to completion of bootstrap. Files attached.
I have been able to run the same docker container on the machine with the mainnet peers lists, etc. without issues since the release with a stop-time of 3 days. First issue observed since the release of the alpha 1.2.1
